opcodes/dxil/dxil_nvapi.cpp

```diff
@@ -317,6 +336,13 @@ bool NVAPIState::commit_opcode(Converter::Impl &impl, bool analysis)
 				return false;
 			break;
 
+		case NV_EXTN_OP_RT_GET_CLUSTER_ID:
+			impl.spirv_module.set_override_spirv_version(0x10400);
```

Comment:
It's too late to do that here. I think during analysis, if NVAPI opcodes are used, just immediate promote to 0x10400 to avoid this problem.

Reply:
I don't quite understand? In practice this works, because [1.4 does end up being written into emitted SPIR-V](https://github.com/HansKristian-Work/dxil-spirv/pull/237/files#diff-6b0fda906f11b50240b80ddd604456312082d76d9110ba7da292bbe4633fa36fR48). Without this (well, more specifically, the overrides below this one), running the tests suite fails with

```
Testing shader: ./shaders/nvapi/ray-query-cluster-id.nvapi.comp
[ERROR]: SPIRV-Tools message: SPV_NV_cluster_acceleration_structure extension requires SPIR-V version 1.4 or later.
  OpExtension "SPV_NV_cluster_acceleration_structure"
```

Reply:
Right. This code path is also taken for analysis, that's true ... Then it's fine as-is.